lib/srv/regular/sshserver.go

```diff
@@ -1264,20 +1264,22 @@ func (s *Server) HandleNewConn(ctx context.Context, ccx *sshutils.ConnectionCont
 	}
 
 	// Create host user.
-	created, userCloser, err := s.termHandlers.SessionRegistry.TryCreateHostUser(identityContext)
+	created, userCloser, err := s.termHandlers.SessionRegistry.UpsertHostUser(identityContext)
 	if err != nil {
-		return ctx, trace.Wrap(err)
+		log.Infof("error while creating host users: %s", err)
 	}
+
 	// Indicate that the user was created by Teleport.
 	ccx.UserCreatedByTeleport = created
 	if userCloser != nil {
 		ccx.AddCloser(userCloser)
 	}
 
-	sudoersCloser, err := s.termHandlers.SessionRegistry.TryWriteSudoersFile(identityContext)
+	sudoersCloser, err := s.termHandlers.SessionRegistry.WriteSudoersFile(identityContext)
 	if err != nil {
-		return ctx, trace.Wrap(err)
+		log.Warnf("error while writing sudoers: %s", err)
```

Comment:
the function only returns unexpected errors, I suggest returning them
```suggestion
		return ctx, trace.Wrap(err, "writing sudoers")
```

Reply:
Tim and I talked about this some last week and decided that we probably shouldn't fail the connection for any error related to host user creation or writing sudoers. Even unexpected ones. We could return the error here but I think we'd just end up ignoring it up the chain where it would be harder to determine which errors are safe to ignore